sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/SdkMeterProvider.java

```diff
@@ -41,10 +35,8 @@ public final class SdkMeterProvider implements MeterProvider, Closeable {
 
   private final ComponentRegistry<SdkMeter> registry;
   private final MeterProviderSharedState sharedState;
-  private final Map<CollectionHandle, CollectionInfo> collectionInfoMap;
```

Comment:
CollectionHandle and CollectionInfo had overlapping responsibilities. I've replaced them with a single class with a name which more clearly conveys its responsibilities: `RegisteredReader`.